tests/shared.test.js

```diff
@@ -0,0 +1,61 @@
+import { configure } from 'enzyme';
+import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
+
+configure({ adapter: new Adapter() });
+
+import React from 'react';
+import { shallow, render } from 'enzyme';
+
+import StarRating from '../src/components/shared/StarRating';
+import Star from '../src/components/shared/Star';
+
+describe('StarRating component', () => {
+  test('should have exactly 5 `Star` components as children', () => {
+    const component = shallow( // mounts a `shallow` render of the component (does not render children)
+      <StarRating rating={3} />
+    );
+    expect(component.find('Star')).toHaveLength(5);
+  });
+});
+
+describe('Star component', () => {
+  test('a Star with with a rating fraction close to 1 should fill 100% of the star', () => {
+    const component = shallow( // mounts a `shallow` render of the component (does not render children)
+      <Star fraction={1} />
+    );
+    expect(component.containsMatchingElement(<stop offset='100%'></stop>)).toBe(true);
+    // expect(component.find('Star')).toHaveLength(5);
```

Comment:
Tests look good-- curious though about the commented out assertion checking for number of stars in each test?

Reply:
Ah! That's a mistake on my part, I had must have forgotten to remove it. It's not supposed to be there.